hubblestack/daemon.py

```diff
@@ -381,21 +387,48 @@ def load_config():
     os.chmod(__opts__['log_file'], 384)
     os.chmod(parsed_args.get('configfile'), 384)
 
+    refresh_grains(initial=True)
+
+    if __salt__['config.get']('hubblestack:splunklogging', False):
+        root_logger = logging.getLogger()
+        handler = hubblestack.splunklogging.SplunkHandler()
+        handler.setLevel(logging.ERROR)
+        root_logger.addHandler(handler)
+
+
+def refresh_grains(initial=False):
+    '''
+    Refresh the grains, pillar, utils, modules, and returners
+    '''
+    global __opts__
+    global __grains__
+    global __utils__
+    global __salt__
+    global __pillar__
+    global __returners__
+    if 'grains' in __opts__:
+        __opts__.pop('grains')
+    if 'pillar' in __opts__:
+        __opts__.pop('pillar')
     __grains__ = salt.loader.grains(__opts__)
+    __grains__['session_uuid'] = SESSION_UUID
     __pillar__ = {}
     __opts__['grains'] = __grains__
     __opts__['pillar'] = __pillar__
     __utils__ = salt.loader.utils(__opts__)
     __salt__ = salt.loader.minion_mods(__opts__, utils=__utils__)
     __returners__ = salt.loader.returners(__opts__, __salt__)
-
-    if __salt__['config.get']('hubblestack:splunklogging', False):
-        hubblestack.splunklogging.__grains__ = __grains__
-        hubblestack.splunklogging.__salt__ = __salt__
-        root_logger = logging.getLogger()
+    hubblestack.splunklogging.__grains__ = __grains__
+    hubblestack.splunklogging.__salt__ = __salt__
+    if not initial and __salt__['config.get']('hubblestack:splunklogging', False):
+        class MockRecord(object):
+            def __init__(self, message, levelname, asctime, name):
+                self.message = message
+                self.levelname = levelname
+                self.asctime = asctime
+                self.name = name
         handler = hubblestack.splunklogging.SplunkHandler()
-        handler.setLevel(logging.ERROR)
-        root_logger.addHandler(handler)
+        handler.emit(MockRecord(str(__grains__), 'INFO', time.asctime(), 'hubblestack.grains_report'))
```

Comment:
Just get rid of the `str()` around `__grains__` and the message will get sent in all extracted and beautiful.

Reply:
Hmmm. I thought I tried that and it didn't work, but maybe something else was going on. Glad it's working for you! I'll do some more testing when we get closer to release as well.